src/briefcase/commands/convert.py

```diff
@@ -0,0 +1,866 @@
+            default = find_most_similar_path_by_name(valid_root_children, module_name)
+
+        intro = f"""
+To set up an existing project for Briefcase, we need to know the path of the
+application entry point relative to the project root (the current working directory).
+
+For example, if you have an existing project, myapp, and you can start myapp by
+running ``src/myapp/__main__.py``, then you should set the source directory to
+``src/myapp``.
+
+Based on your project's folder layout, we believe it might be {default}."""
+        return default, intro
+
+    def input_source_dir(self, app_name: str) -> str:
+        """Ask about the source dir, using hints from the project folder layout.
+
+        :returns: The source directory
+        """
+
+        default, intro = self.get_source_dir_hint(app_name)
+        return self.input_text(
+            intro=intro,
+            variable="source directory",
+            default=default,
+            validator=self.validate_source_dir,
+        )
+
+    def input_test_source_dir(self, app_name) -> str:
+        """Ask about the test source dir, using hints from the project folder layout.
+
+        :returns: The test source directory
+        """
+        intro = """
+We also need to know the path to the test suite (if it exists). The test path
+should be relative to the project root directory.
+
+If the provided directory doesn't exist, it will be created and populated with
+some default test files."""
+        if Path("test").exists():
+            default = "test"
+            intro += (
+                "\n\nBased on your project's folder structure, we believe "
+                + '"test" might be your test directory'
+            )
+        elif Path("tests").exists():
+            default = "tests"
+            intro += (
+                "\n\nBased on your project's folder structure, we believe "
+                + '"tests" might be your test directory'
+            )
+        else:
+            default = "tests"
+
+        return self.input_text(
+            intro=intro,
+            variable="test source directory",
+            default=default,
+            validator=self.create_test_source_dir_validator(app_name=app_name),
+        )
+
+    def input_description(self) -> str:
+        """Ask about the app description, using hints from the pyproject.toml file if
+        there are any.
+
+        :returns: The app description
+        """
+        intro = """
+Now, we need a one line description for your application."""
+
+        if "description" in self.pep621_data:
+            default = self.pep621_data["description"]
+            intro += " Based on the PEP-621 formatted pyproject.toml file, we believe it could be"
+            intro += f"\n'{default}'"
+        else:
+            default = "My first application"
+
+        description = self.input_text(
+            intro=intro,
+            variable="description",
+            default=default,
+        )
```

Comment:
We don't need *both* a PEP621 description and a `tool.briefcase.description`. If they've got PEP621 metadata, we can use it as-is and omit that key from the `tool.briefcase` section.

Reply:
Hmmm, a solution to that might be changing the template so if the description field is empty, then it's skipped?

Reply:
I guess it depends what impact that has on the "merge" part of the process. Preference will always be for a simpler template; but if having a slightly more complicated template makes the merge process significantly easier, then that seems like a reasonable price to pay. However, my initial impression is that it should be possible to put a dummy value in place, and then exclude the briefcase-level description if there's a PEP621 description during the process of merging the metadata.

Reply:
I guess this is irrelevant now that Briefcase also reads PEP621 formatted pyproject.toml files? We updated the merging so overlapping fields are removed if they are equal.

Reply:
Agreed, skipping unambiguous fields is a good idea. We've tried to remove the prompts for the unambiguous fields now. Do you think we should show a message along the lines of

```raw
We need a name that can serve as a machine-readable Python package name for your application.
This name must be PEP508-compliant - that means the name may only contain letters, numbers, hyphens
and underscores; it can't contain spaces or punctuation, and it can't start with a hyphen or underscore.

Based on your PEP621 formatted pyproject.toml file, we assume it is {app_name}.
```

(and similar for the description)? Or is it fine to just skip to the next field?

Also, the way we delete overlapping fields is with `tomli_w`, but as you mentioned here: https://github.com/beeware/briefcase/pull/1231#pullrequestreview-1830961522, this approach doesn't preserve formatting. Alternatively, since all the fields we remove are single-line fields, we could delete the lines manually by checking e.g. `line.startswith("description=")` etc, but that might be a bit less fault tolerant? What do you think?

Reply:
> Agreed, skipping unambiguous fields is a good idea. We've tried to remove the prompts for the unambiguous fields now. Do you think we should show a message along the lines of ... (and similar for the description)? Or is it fine to just skip to the next field?

I think the approach should mirror what happens with project_overrides - we don't need to echo the explanatory text, but we *should* indicate the value that was selected.

> Also, the way we delete overlapping fields is with `tomli_w`, but as you mentioned here: [#1231 (review)](https://github.com/beeware/briefcase/pull/1231#pullrequestreview-1830961522), this approach doesn't preserve formatting. Alternatively, since all the fields we remove are single-line fields, we could delete the lines manually by checking e.g. `line.startswith("description=")` etc, but that might be a bit less fault tolerant? What do you think?

I agree that a pattern match approach will be less fault tolerant, so that's not the right way to go.

There is another option - ignore the problem. Having a duplicated key isn't actually an issue - Briefcase will always use the "most specific" entry; it's mildly undesirable from a "don't repeat yourself" perspective, but given the alternatives, and the specific circumstances (migrating an existing project), maybe that's an acceptable price?